plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobContainerRetriesTests.java

```diff
@@ -294,6 +296,44 @@ public void testWriteLargeBlob() throws Exception {
         assertThat(blocks.isEmpty(), is(true));
     }
 
+    public void testRetryUntilFail() throws IOException {
```

Comment:
This test is admittedly a little random/weird, but I figured it captured the initial issue that showed this bug to some degree at least.